File: ontrack-model/src/main/java/net/nemerosa/ontrack/model/support/ConfigurationProperty.java
package net.nemerosa.ontrack.model.support;

/**
 * Type of {@linkplain net.nemerosa.ontrack.model.structure.PropertyType property} which gives access to
 * a {@link UserPasswordConfiguration configuration}.
 *
 * @param <T> Type of the configuration
 */
public interface ConfigurationProperty<T extends UserPasswordConfiguration<T>> {

    /**
     * Access to the configuration.
     */
    T getConfiguration();

}

File: ontrack-model/src/main/java/net/nemerosa/ontrack/model/support/ConfigurationProperty.kt
package net.nemerosa.ontrack.model.support

/**
 * Type of [property][net.nemerosa.ontrack.model.structure.PropertyType] which gives access to
 * a [configuration][Configuration].
 *
 * @param <T> Type of the configuration
 */
interface ConfigurationProperty<T : Configuration<T>> {

    /**
     * Access to the configuration.
     */
    val configuration: T

}